packages/contracts-bedrock/src/L2/SuperchainERC20.sol

```diff
@@ -0,0 +1,147 @@
+    event SentERC20(address indexed from, address indexed to, uint256 amount, uint256 chainId, bytes data);
+
+    /// @notice Emitted whenever tokens are successfully relayed on this chain.
+    /// @param to     Address of the recipient.
+    /// @param amount Amount of tokens relayed.
+    /// @param data   Data sent with the message.
+    event RelayedERC20(address indexed to, uint256 amount, bytes data);
+
+    /// @notice A modifier that only allows the bridge to call
+    modifier onlyBridge() {
+        if (msg.sender != BRIDGE) revert CallerNotBridge();
+        _;
+    }
+
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0
+    string public constant version = "1.0.0";
+
+    /// @param _bridge      Address of the L2 standard bridge.
+    /// @param _name        ERC20 name.
+    /// @param _symbol      ERC20 symbol.
+    /// @param _decimals    ERC20 decimals.
+    constructor(address _bridge, string memory _name, string memory _symbol, uint8 _decimals) ERC20(_name, _symbol) {
+        BRIDGE = _bridge;
+        DECIMALS = _decimals;
+    }
+
+    /// @notice Allows the StandardBridge on this network to mint tokens.
+    /// @param _to     Address to mint tokens to.
+    /// @param _amount Amount of tokens to mint.
+    function mint(address _to, uint256 _amount) external virtual override onlyBridge {
+        _mint(_to, _amount);
+        emit Mint(_to, _amount);
+    }
+
+    /// @notice Allows the StandardBridge on this network to burn tokens.
+    /// @param _from   Address to burn tokens from.
+    /// @param _amount Amount of tokens to burn.
+    function burn(address _from, uint256 _amount) external virtual override onlyBridge {
+        _burn(_from, _amount);
+        emit Burn(_from, _amount);
+    }
+
+    /// @notice Sends tokens to some target address on another chain.
+    /// @param _to      Address to send tokens to.
+    /// @param _amount  Amount of tokens to send.
+    /// @param _chainId Chain ID of the destination chain.
+    /// @param _data    Data to be sent with the message.
+    function sendERC20(address _to, uint256 _amount, uint256 _chainId, bytes memory _data) external override {
+        _burn(msg.sender, _amount);
+
+        bytes memory _message = abi.encodeCall(this.relayERC20, (_to, _amount, _data));
+        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
+
+        emit SentERC20(msg.sender, _to, _amount, _chainId, _data);
+    }
+
+    /// @notice Relays tokens received from another chain.
+    /// @param _to     Address to relay tokens to.
+    /// @param _amount Amount of tokens to relay.
+    /// @param _data   Data sent with the message.
+    function relayERC20(address _to, uint256 _amount, bytes memory _data) external override {
+        if (msg.sender != MESSENGER) revert RelayMessageCallerNotL2ToL2CrossDomainMessenger();
+
+        if (IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSender() != address(this)) {
```

Comment:
Even though is not the same case, we were talking about following the OP coding style. In this case we could use this approach... or not.

https://github.com/ethereum-optimism/optimism/blob/acafdb9eed71d07dd9dbf3944195840cbd61fd8b/packages/contracts-bedrock/src/universal/StandardBridge.sol#L107-L113

Reply:
agree, it is a more readable change